### DISCUSSION TITLE
Filtering related models

### DISCUSSION THREAD

**Discussion by maslyankov** [collerek/ormar#391] [Q&A]
Hi!
I am trying to find a way to get all of a model's ForeignKey relations (OneToMany relations) and filter them.

Use case:
```python
today: datetime.datetime = datetime.date.today()
today_start, today_end = (datetime.datetime.combine(today, datetime.time(00, 00, 00)).timestamp(),
                          datetime.datetime.combine(today, datetime.time(23, 59, 59)).timestamp())
area_rooms = await app.core.models.Room.objects.filter(lsb_area=area_id).select_related("entries").filter(
        (app.core.models.Room.entries.start_time >= today_start) & (app.core.models.Room.entries.end_time <= today_end)
    ).all()
```

As far as I understand, this filters the parent model, not the child models? Is there a way to achieve this with similar syntax?
Thank you!

**Comment by collerek** [comment]
In queries without limit/offset the filters are applied on the joined result.
In your sample, that means that result will contain only rooms from given area that have entries in given time interval.

That means that you will have:
- Only rooms that have entries in given time interval (so filtered parent)
- With only children from given interval (filtered children) - i.e. if a room have more entires outside of filter they will not be fetched (so filtered children).

**Reply by maslyankov** [reply]
Thank you for the support! How would you suggest to fetch all rooms but only filter the children they have?
